src/main/java/spoon/support/visitor/java/JavaReflectionVisitorImpl.java

```diff
@@ -46,38 +45,70 @@ public void visitPackage(Package aPackage) {
 		if (clazz.getPackage() != null) {
 			clazz.getPackage();
 		}
-		for (TypeVariable<Class<T>> generic : clazz.getTypeParameters()) {
-			visitTypeParameter(generic);
+		try {
+			for (TypeVariable<Class<T>> generic : clazz.getTypeParameters()) {
+				visitTypeParameter(generic);
+			}
+		} catch (Throwable ignore) {
+			// partial classpath
 		}
-		if (clazz.getGenericSuperclass() != null && clazz.getGenericSuperclass() != Object.class) {
-			visitTypeReference(CtRole.SUPER_TYPE, clazz.getGenericSuperclass());
+		try {
+			if (clazz.getGenericSuperclass() != null && clazz.getGenericSuperclass() != Object.class) {
+				visitTypeReference(CtRole.SUPER_TYPE, clazz.getGenericSuperclass());
+			}
+		} catch (Throwable ignore) {
+			// partial classpath
 		}
-		for (Type anInterface : clazz.getGenericInterfaces()) {
-			visitTypeReference(CtRole.INTERFACE, anInterface);
+		try {
+			for (Type anInterface : clazz.getGenericInterfaces()) {
+				visitTypeReference(CtRole.INTERFACE, anInterface);
```

Comment:
I suggest to try/catch this line only. So some interfaces are correctly created and some are ignored. Actually it process all interfaces until first which is not on classpath.

Do the same in all cycles of this method, please.

Reply:
I cannot the Error is thrown by clazz.getGenericInterfaces()

Reply:
and I did not succeed to find a way to get a partial list :/

Reply:
So then try/catch only clazz.getGenericSuperclass(). So the exceptions thrown in visitTypeReference are not catched

Reply:
I don't have the motivation to do it, right now. 
There is probably 50 invocations to protect in this class.

Reply:
it can be written in nice way using lambda:
`for (Type anInterface : ignoreMissingClass(()->clazz.getGenericInterfaces()))`
where ignoreMissingClass will return empty collection in case of exception

Reply:
clazz.getGenericInterfaces() returns an array not a collection.
I don't see a way to do it nicely :/

Reply:
but the problem is to create EMPTY_ARR, because EMPTY_ARR has to be different for each T